whisper/whisperv5/peer.go

```diff
@@ -113,7 +117,27 @@ func (p *Peer) update() {
 				log.Trace("broadcast failed", "reason", err, "peer", p.ID())
 				return
 			}
-
+		case <-hashesTransmit.C:
+			if err := p.broadcastHashes(hashes); err != nil {
+				log.Trace("broadcast of hashes failed", err, "peer", p.ID)
+				return
+			}
+			hashes = hashes[:0]
+		case hash := <-p.hashes:
+			if _, ok := p.advertised[hash]; ok {
+				continue
+			}
+			if p.known.Has(hash) {
```

Comment:
I kind of see why you maintain two tables (`known` and `advertised`) but I wonder if it wouldn't be better to have a list of hashes "recently added" that will make it to the next "advertisement campaign", to be emptied after each send ?